x-pack/plugins/apm/public/components/app/metrics_details/service_node_metrics/index.tsx

```diff
@@ -83,9 +85,17 @@ export function ServiceNodeMetrics({ serviceNodeName }: Props) {
     environment,
   });
 
+  const preferred = usePreferredDataSourceAndBucketSize({
+    start,
+    end,
+    kuery,
+    type: ApmDocumentType.TransactionMetric,
```

Comment:
I think this should be `ServiceTransactionMetric`
@sqren Can you confirm ?

Reply:
@sqren , @achyutjhunjhunwala how I know the number of metrics and the preferred document type I should pass in each endpoint?

Reply:
Also, what I should do in cases like this one
`    apm: {
      events: [
        getProcessorEventForTransactions(searchAggregatedTransactions),
        ProcessorEvent.error,
        ProcessorEvent.metric,
      ],
    },`
    
How we replace `getProcessorEventForTransactions(searchAggregatedTransactions)`?

Reply:
> how I know the number of metrics

I believe you mean `numBuckets` parameter. 
Rule of thumb is 20 for Spark Plots, 100 for every other chart. But higher number also means high performance cost. Now, `numBuckets` is used to calculate `bucketSizeInSeconds` which you don't need in all the cases. So like for this example, you can send 20 also.

> and the preferred document type

if the data we are are trying to get is available in ServiceTransactionMetrics, then that becomes the choice. One expectation from this task is also to migrate certain places where we are still using `TransactionMetrics` to `ServiceTransactionMetrics`, but yes its not in every place.

@sqren @dgieselaar can confirm this better